pkg/logql/log/dedup_filter.go

```diff
@@ -0,0 +1,59 @@
+package log
+
+import (
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+type LineDedupFilter struct {
+	labels     map[string]interface{}
+	inverted   bool
+	hashLookup map[uint64]struct{}
+}
+
+func NewLineDedupFilter(labelFilters []string, inverted bool) *LineDedupFilter {
+	// create a map of labelFilters for O(1) lookups instead of O(n)
+	var filterMap = make(map[string]interface{})
```

Comment:
nit: can use `map[string]struct{}` here.

Reply:
Ah yes, I missed this in my refactor - thanks :+1